scripts/yyVariable.js

```diff
@@ -614,25 +658,14 @@ function array_insert( _array, _index )
 {
     if (Array.isArray(_array)) {
         _index = yyGetInt32(_index);
-        if(_index > _array.length)
-        {
-            var currLen = _array.length;
-            var n;
-            for(n = 2; n < arguments.length; ++n) {
-                _array[_index+n-2] = arguments[n];
-            }
-            // zero fill back to original length
-            for(n = _index - 1; n >= currLen; --n)
-            {
-                _array[n] = 0;
-            }
-        }
-        else
-        {
-            for( var n=2; n<arguments.length; ++n) {
-                _array.splice( _index+n-2, 0, arguments[n] );
-            } // end for
+
+        // Fill any null elements prior to the copy with 0 (real) values. 
+        // This mimics the initialisation flow of array data on VM/YYC when MemoryManager::SetLength is called. 
+        for(n = _index - 1; n >= _array.length; --n) { 
+            _array[n] = 0; 
         }
+
+        _array.splice(_index, 0, ...Array.prototype.slice.call(arguments, 2));
     } // end if
     else {
         yyError( "argument0 is not an array");

```

Comment:
Could do with some of these existing error messages also having the function name on them?